Week5/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/activities/MainActivity.kt

```diff
@@ -13,22 +17,48 @@ import kotlinx.android.synthetic.main.activity_main.*
 class MainActivity : AppCompatActivity(), MovieAdapter.MovieDetailListener {
     companion object {
         const val MOVIE_DETAILS = "movie_details"
+        const val IS_LOGGED_IN = "isLoggedIn"
+        const val LOGGED_IN_STATUS_KEY = "loggedInStatus"
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        moviesRecyclerView.layoutManager = LinearLayoutManager(this)
-        moviesRecyclerView.adapter =
-            MovieAdapter(
-                DummyData.movies,
-                this
-            )
+
+        val sharedPrefs =
+            getSharedPreferences(LOGGED_IN_STATUS_KEY, Context.MODE_PRIVATE)
+        val isLoggedIn = sharedPrefs.getBoolean(IS_LOGGED_IN, false)
```

Comment:
Same thing here, too mch code in onCreate, should be split up. Also, I'm not sure how you can reach the MainActivity, and *not* be logged in. And also, if you start Login, and not finish this activity, the user can just tap back, and enter the Main Activity again, without triggering onCreate.